lib/rules/no-extra-parens.js

```diff
@@ -418,6 +418,7 @@ module.exports = {
         function checkExpressionOrExportStatement(node) {
             const firstToken = isParenthesised(node) ? sourceCode.getTokenBefore(node) : sourceCode.getFirstToken(node);
             const secondToken = sourceCode.getTokenAfter(firstToken, astUtils.isNotOpeningParenToken);
+            const thirdToken = secondToken ? sourceCode.getTokenAfter(secondToken) : {};
```

Comment:
Nitpick: Instead of defaulting the token to an empty object, can you add  a null check to the conditional?

```js
/* ... */ && thirdToken && thirdToken.type === "Keyword" && /* ... */
```

I'm worried it would be confusing for a variable to sometimes reference a `Token` and sometimes reference an empty object.